x/sequencer/controller/store.go

```diff
@@ -74,17 +74,26 @@ func (store *Store) processPayload(payload *listener.Payload) {
 	store.mtx.Unlock()
 }
 
-func transactions(payload *listener.Payload) []*types.ArweaveTransaction {
+func (store *Store) transactions(payload *listener.Payload) []*types.ArweaveTransaction {
 	txs := make([]*types.ArweaveTransaction, 0, len(payload.Transactions))
 	for _, tx := range payload.Transactions {
+		err := warp.ValidateTags(tx.Tags)
+		if err != nil {
+			store.Log.WithError(err).WithField("tx_id", tx.ID).Warn("Ignored interaction with invalid tags")
+			continue
+		}
+
 		contract := getContractFromTag(tx)
-		if contract != "" {
-			txs = append(txs, &types.ArweaveTransaction{
-				Id:       tx.ID.Base64(),
-				Contract: contract,
-				SortKey:  warp.CreateSortKey(tx.ID, payload.BlockHeight, payload.BlockHash),
-			})
+		if len(contract) == 0 {
+			store.Log.WithField("tx_id", tx.ID).Warn("Ignored interaction without a contract tag")
```

Comment:
tx.ID will print a table of bytes instead, maybe use tx.ID.Base64()

Reply:
Fixed